src/connection.js

```diff
@@ -689,17 +689,22 @@ class Connection extends EventEmitter {
       if (err) {
         return this.socketError(err);
       }
+      if (this.state === this.STATE.FINAL) {
+        if (socket) {
```

Comment:
I'm pretty sure `socket` is always set here.

Can we change the code to do an early return instead?

```js
if (this.state === this.STATE.FINAL) {
  socket.destroy();
  return;
}

this.socket = socket;
...
```

Also, could you add an empty newline before the `if` statement here?